### ISSUE TITLE
Unexpected indentation error

### ISSUE THREAD

**Issue by amk221** [eslint/eslint#8771]
<img width="531" alt="screen shot 2017-06-21 at 10 45 54" src="https://user-images.githubusercontent.com/382061/27390580-8f8d04e2-5699-11e7-9d0d-49748604ed9a.png">

With `indent: ['error', 2]` I wasn't expecting this ^

If I put the Mixins on a single line or remove them, the problem goes away

v4.0.0

**Comment by not-an-aardvark** [comment]
Thanks for the report. This is working as intended -- your `init` function is nested in an object, so the `indent` rule expects it to be offset by 2 spaces from the start of the object.

**Comment by amk221** [comment]
I thought that might be the case, any idea how I can get it to work (with a custom rule, if need be)

Because the following examples, although technically correct, look wrong to the eye:

```javascript
// technically correct, but looks wrong
export default Component.extend(
Mixin1,
Mixin2,
Mixin3, {

  init() {

  }
});
```

```javascript
// technically correct, but also looks wrong
export default Component.extend(
  Mixin1,
  Mixin2,
  Mixin3, {

    init() {

    }
  });
```

```javascript
// fails, but looks best
export default Component.extend(
  Mixin1,
  Mixin2,
  Mixin3, {

  init() {

  }
});
```

**Comment by not-an-aardvark** [comment]
At the moment, you would have to disable the entire `indent` rule to get this to pass. However, when https://github.com/eslint/eslint/issues/8594 is added it will be possible to configure the `indent` rule to ignore only the indentation of your `init` method, while still checking everything else (and all the code inside the method). If you wanted to, you could then also write a custom rule to verify that your `init` method has the indentation you prefer.

**Comment by amk221** [comment]
Thanks
